internal/profiles/service.go

```diff
@@ -47,45 +47,29 @@ import (
 type ProfileService interface {
 	// CreateProfile creates the profile in the specified project
 	// returns the updated profile structure on successful update
+	// subscriptionID should be set to nil when not calling
 	CreateProfile(
-		ctx context.Context,
```

Comment:
As part of this work, I decided to simplify the interface for profiles and rules to get rid of the dedicated subscription methods - now there is a single method and you pass a nil UUID if you do not need a subscription to be set.

Reply:
Do you suppose we will move more database methods here, e.g. Delete? If we do do you think we should rather have CreateWithTransaction accepting the db.ExtendQuerier interface and the simple functions being wrappers around the WithTransaction using the db.WithTransaction wrapper?